### ISSUE TITLE
allow reset API to bring new signals as atomic operation

### ISSUE THREAD

**Issue by longquanzheng** [temporalio/features#553]
### Describe the solution you'd like

It would be really nice to allow reset API to bring a [few] new signals as atomic operation.
I can see several use cases for that
reset and then skip some activity that we don’t want to rerun. We could add some simple code to skip based on the signal. This will be much simpler than any other alternatives that I have seen.
reset with cherry pick some signals but not all. Currently we have the option to preserve all signals after reset point, or none. But it’s a hard decision when we want to preserve some.
Reset with some special behavior— just let the workflow knows that current run is in a reset state so that it can adjust the behavior accordingly.
Etc etc

**Comment by longquanzheng** [comment]
> This may be similar to #1203

Yeah that looks very similar: https://github.com/temporalio/temporal/issues/1203

**Comment by tconley1428** [comment]
Duplicate of the issue above, now moved to discussions at https://github.com/temporalio/temporal/discussions/12161.
